client/src/GameState.ts

```diff
@@ -95,6 +106,32 @@ export class GameState {
                 this.emitAndPlaceCurrentTetromino();
             }
         });
+        this.socket.on("playerTrade", (playerId, _, otherTradeState) => {
+            console.log("Request received");
+            if (otherTradeState == TradeState.Offered) {
+                this.tradeState = TradeState.Pending;
+                this.tradingPlayerId = playerId;
+            } else if (otherTradeState == TradeState.Accepted) {
+                this.tradeState = TradeState.NoTrade;
+                this.tradingPlayerId = null;
+            }
+        });
+        this.socket.on("sendTradePiece", (tetrominoType) => {
+            this.currentTetromino.swapPiece(tetrominoType);
+            this.currentTetromino.isTraded = true;
+            this.clearTradeAndEmit();
```

Comment:
Should we also be calling `this.emitPlayerMove();` to update everyone with the new position?

Reply:
I guess this is probably necessary since position has changed. It's been added.